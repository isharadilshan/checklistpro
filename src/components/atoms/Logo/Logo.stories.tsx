import React from 'react';
import {NativeBaseProvider} from 'native-base';
import {ComponentMeta, ComponentStory} from '@storybook/react';

import Logo from './';

export default {
  title: 'components/atoms/Logo',
  component: Logo,
} as ComponentMeta<typeof Logo>;

export const Basic: ComponentStory<typeof Logo> = () => (
  <NativeBaseProvider>
    <Logo />
  </NativeBaseProvider>
);
