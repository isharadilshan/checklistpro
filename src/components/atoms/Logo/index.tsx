import React, { memo } from 'react';
import { Image } from 'react-native';
import createStyle from './style';
import { Center } from 'native-base';

const Logo: React.FC = () => {
  const styles = createStyle();
  return (
    <Center style={styles.wrapper}>
      <Image
        source={require('../../../assets/logo/logo.png')}
        style={styles.image}
      />
    </Center>
  );
};

export default memo(Logo);
