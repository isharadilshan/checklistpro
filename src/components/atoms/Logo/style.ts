import {ImageStyle, StyleSheet, ViewStyle} from 'react-native';
interface Style {
  wrapper: ViewStyle;
  image: ImageStyle;
}

export default () => {
  return StyleSheet.create<Style>({
    wrapper: {
      marginVertical: 100,
    },
    image: {
      width: 128,
      height: 128,
      marginBottom: 12,
    },
  });
};
