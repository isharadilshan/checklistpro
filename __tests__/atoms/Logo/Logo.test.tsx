import React from 'react';
import {describe, expect} from '@jest/globals';
import {shallow, configure} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';

import Logo from '../../../src/components/atoms/Logo';

configure({adapter: new Adapter()});

const props = {
  style: {width: 200, height: 150},
};

describe('when rendering', () => {
  const wrapper = shallow(<Logo {...props} />);

  it('should render a component matching the snapshot', () => {
    const tree = toJson(wrapper);
    expect(tree).toMatchSnapshot();
    expect(wrapper).toHaveLength(1);
  });
});
