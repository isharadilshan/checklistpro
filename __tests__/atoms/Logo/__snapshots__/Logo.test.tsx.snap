// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when rendering should render a component matching the snapshot 1`] = `
<Image
  source={
    {
      "testUri": "../../../src/assets/logo/swivel_tech.png",
    }
  }
  style={
    {
      "height": 128,
      "marginBottom": 12,
      "width": 128,
    }
  }
/>
`;
